integration/test_dynamic_links.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Integration tests for firebase_admin.auth module."""
+import sys
+
+import pytest
+import requests
+
+
+from firebase_admin import dynamic_links
+
+from tests import testutils
+
+dynamic_links_e2e_url = ''
+try:
+    dynamic_links_e2e_url = testutils.resource('dynamic_links_e2e_url.txt').strip()
+except IOError:
+    sys.stderr.write("""
+==============================================================================================
+    To run end to end tests you must do the following:
+        1. From the firebase console, create a short link under dynamic links.
+        2. From your broser or phone, go to that short link and see that it redirects.
+        3. Wait up to 36 hours.
+        4. Create a file named dynamic_links_e2e_url.txt under tests/data/.
+        5. Populate that file with the quoted link you created in 1 on the first and only line 
+           e.g. $ echo "https://your1.app.goo.gl/suff" > tests/data/dynamic_links_e2e_url.txt
+===============================================================================================
+""")
+
+class TestEndToEnd(object):
+    """Runs an end to end test, see comment string for setup."""
+
+    def test_get_stats(self):
+        if(dynamic_links_e2e_url) == 0:
+            return
```

Comment:
Use the pytest `skipif` annotation: https://docs.pytest.org/en/2.9.0/skipping.html

it can be applied to an entire class/module too